core/message_parsers.py

```diff
@@ -0,0 +1,350 @@
+        
+        # Format ticks
+        plt.xticks(range(len(dates)), dates, rotation=45)
+        ax.tick_params(colors='white')
+        
+        # Add labels
+        plt.ylabel('Experience', color='white')
+        plt.title('Daily Guild Experience Trend', color='white', pad=20)
+        
+        # Adjust layout
+        plt.tight_layout()
+        
+        # Save to bytes buffer
+        buf = io.BytesIO()
+        plt.savefig(buf, format='png', dpi=100, bbox_inches='tight')
+        buf.seek(0)
+        plt.close()
+        
+        return buf
+
+    def _parse_guild_data(self, data_string):
+        guild_data = {}
+    
+        lines = data_string.strip().split('\n')
+        
+        # Remove the first line if it's empty or a header
+        if lines and (not lines[0].strip() or ':' not in lines[0]):
+            lines = lines[1:]
+        
+        for line in lines:
+            line = line.strip()
+            if not line:
+                continue
+            
+            if ':' in line:
+                key, value = line.split(':', 1)
+                key = key.strip()
+                value = value.strip()
+                
+                if key == 'Created':
+                    guild_data['created'] = value
+                elif key == 'Members':
+                    guild_data['members'] = value
+                elif key == 'Guild Exp':
+                    exp_rank = value.split()
+                    guild_data['total_exp'] = exp_rank[0]
+                    guild_data['rank'] = exp_rank[1].strip('()')
+                elif key == 'Guild Level':
+                    guild_data['level'] = value
+                else:
+                    # Parse daily experience entries
+                    if any(x in key for x in ['Today', 'Oct']):
+                        if 'daily_exp' not in guild_data:
+                            guild_data['daily_exp'] = []
+                        exp_value = int(value.split()[0].replace(',', ''))
+                        guild_data['daily_exp'].append((key.strip(), exp_value))
+        
+        return guild_data
+
+    def _create_guild_stats_embed(self):
+        input_data = self.raw_message
+        # Parse the input data
+        guild_data = self._parse_guild_data(input_data)
+        
+        # Create the main description
+        description = f"""
+    ## Guild Stats and Experience History
+
+    • **Created:** {guild_data.get('created', 'Unknown')}
+    • **Members:** {guild_data.get('members', 'Unknown')}
+    • **Guild Level:** {guild_data.get('level', 'Unknown')}
+    • **Total Experience:** {guild_data.get('total_exp', 'Unknown')} **{guild_data.get('rank', '')}**
+    """
+        
+        # Create the embed
+        embed = discord.Embed(
+            description=description,
+            color=0x5865F2  # Discord blurple color
+        )
+        
+        # Create the graph
+        graph_buffer = self._create_exp_graph(guild_data['daily_exp'])
```

Comment:
This call is almost certainly blocking for a few ms, can we wrap it in an executor?

Reply:
I will try it out

Reply:
done